src/nrfx/hal/nrf_clock.h

```diff
@@ -280,6 +309,61 @@ static inline nrf_clock_lfclk_t nrf_clock_lf_src_get(NRF_CLOCK_Type const * p_re
 {
     return (nrf_clock_lfclk_t)(p_reg->LFCLKSRC);
 }
+
+nrf_clock_is_running(NRF_CLOCK_Type const * p_reg, nrf_clock_domain_t domain,
+                    void * clk_src)
+{
+    switch (domain)
+    {
+        case NRF_CLOCK_DOMAIN_LFCLK:
+            if (clk_src != NULL)
+            {
+                (*(uint32_t *)clk_src) = ((p_reg->LFCLKSTAT & CLOCK_LFCLKSTAT_SRC_Msk)
+                                          >> CLOCK_LFCLKSTAT_SRC_Pos);
+            }
+            if ((p_reg->LFCLKSTAT & CLOCK_LFCLKSTAT_STATE_Msk)
+                >> CLOCK_LFCLKSTAT_STATE_Pos)
+            {
+                return true;
+            }
+            break;
+        case NRF_CLOCK_DOMAIN_HFCLK:
+            if (clk_src != NULL)
+            {
+                (*(uint32_t *)clk_src) = ((p_reg->HFCLKSTAT & CLOCK_HFCLKSTAT_SRC_Msk)
+                                          >> CLOCK_HFCLKSTAT_SRC_Pos);
+            }
+            if ((p_reg->HFCLKSTAT & CLOCK_HFCLKSTAT_STATE_Msk)
+                >> CLOCK_HFCLKSTAT_STATE_Pos)
```

Comment:
If you're not going to compare the shifted result against something, why shift it?  Unless the `_Msk` and `_Pos` are wrong this is functionally equivalent to testing `p_reg->HFCLKSTAT & CLOCK_HFCLKSTAT_STATE_Msk` for a non-zero value.